### ISSUE TITLE
Review request for Confirmation of Action API

### ISSUE THREAD

**Issue by sartang** [w3ctag/design-reviews#713]
Braw mornin' TAG!
 
I'm requesting a TAG review of a Confirmation of Action API.
 
ARIA live regions are the only mechanism available today that communicate content changes down to the accessibility layer so that users can hear about them. ARIA live regions are stretched far beyond their original use cases as authors struggle to use them in scenarios that they weren't designed for. We propose a notification API purpose-built to communicate to the accessibility layer for scenarios in which ARIA live regions are a poor choice. One of these scenarios is a "confirmation of action" where the action in question is not necessarily tied to UI (elements) in the app.
 
  - Explainer¹ (minimally containing user needs and example code): [here](https://github.com/WICG/aom/blob/gh-pages/notification-api.md)
  - User research: [here](https://github.com/w3c/aria/issues/832)
  - Primary contacts (and their relationship to the specification):
      - [Sara Tang](@sartang), Microsoft
      - [Daniel Libby](@dlibby), Microsoft
      - [Travis Leithead](@travisleithead), Microsoft
  - Organization(s)/project(s) driving the specification: Microsoft, Google, Chrome
  - External status/issue trackers for this specification (publicly visible, e.g. Chrome Status): [here](https://chromestatus.com/feature/5745430754230272)
 
Further details:
 
  - [x] I have reviewed the TAG's [Web Platform Design Principles](https://www.w3.org/TR/design-principles/)
  - Relevant time constraints or deadlines: [please provide]
  - The group where the work on this specification is currently being done: [ARIA working group](https://www.w3.org/groups/wg/aria)
  - The group where standardization of this work is intended to be done (if current group is a community group or other incubation venue):
  - Major unresolved issues with or opposition to this specification:
  - This work is being funded by: Microsoft, Google
 
We'd prefer the TAG provide feedback as (please delete all but the desired option):
 
  💬 leave review feedback as a **comment in this issue** and @-notify [sartang, dlibby]

**Comment by torgo** [comment]
Hi @sartang - thanks for this really well-written explainer! We especially appreciate how it starts with the user needs.   Do you have any further info on multi-stakeholder interest?   Have they had feedback from the ARIA community on this?

**Comment by sartang** [comment]
@torgo Not at the moment, but we are hoping to get more feedback once we start prototyping.

**Comment by dlibby-** [comment]
As @sartang mentioned, we're planning on getting more feedback from other stakeholders as we refine the proposal during implementation and modification to the proposal.

We have received some preliminary feedback from various folks (see [initial discussion](https://docs.google.com/document/d/1dawqIsbwcHp1eGmQGTkKws_K1a7-MuIBVgFz3VmwPsQ/edit#heading=h.dpvsjlsvvw28) in the AOM CG and https://github.com/WICG/aom/issues/187), and once we believe we've addressed that feedback, we plan on opening this up for discussion in ARIA WG.

cc @travisleithead in case he wanted to add any details.

**Comment by LeaVerou** [comment]
Hi there,

We looked at this today during our F2F. Overall this looks like useful functionality that has been missing from the web platform. We had a question: why hang `ariaNotify()` off `document.body` which doesn't exist in contexts other than HTML (e.g. SVG)? Why not e.g. `document.documentElement` which exists everywhere?

**Comment by LeaVerou** [comment]
Hi there,

We looked at this today during our F2F. Overall this looks like useful functionality that has been missing from the web platform. We'd love to see this moving forward. Thanks for working with us!

**Comment by sartang** [comment]
Leaving a note here that I've updated the explainer. The proposal is largely the same, but more code examples have been added, the parameter names in the spec have changed slightly, and a section detailing how to fallback / transition the feature.
